.github/workflows/build-test-core-win64.yml

```diff
@@ -0,0 +1,81 @@
+# AUTOGENERATED FROM build-test-core-win64.jsonnet DO NOT MODIFY
+name: build-test-core-win64
+"on":
+  workflow_dispatch:
+  workflow_call:
+jobs:
+  job:
+    runs-on: windows-latest
+    defaults:
+      run:
+        shell: bash
+        working-directory: checkout
+    steps:
+    - name: Configure git
+      working-directory: ""
+      run: |
+        git config --system core.longpaths true
+        git config --global fetch.parallel 50
+    - uses: actions/checkout@v3
+      with:
+        submodules: true
+        path: checkout
+    - uses: ocaml/setup-ocaml@v2
+      with:
+        ocaml-compiler: "4.14"
+        opam-pin: false
+        allow-prerelease-opam: true
+        opam-repositories: |
+          opam-repository-mingw: https://github.com/ocaml-opam/opam-repository-mingw.git#sunset
+          default: https://github.com/ocaml/opam-repository.git
+    - name: Build tree-sitter
+      env:
+        CC: x86_64-w64-mingw32-gcc
+      run: |
+        cd libs/ocaml-tree-sitter-core
+        ./configure
+        ./scripts/download-tree-sitter --lazy
+        prefix="$(pwd)/tree-sitter"
+        cd downloads/tree-sitter
+        make PREFIX="$prefix" CFLAGS="-O3 -Wall -Wextra"
+        make PREFIX="$prefix" install
+    - name: Install deps
+      run: |
+        export PATH="${CYGWIN_ROOT_BIN}:${PATH}"
+        opam depext conf-pkg-config conf-gmp conf-libpcre
+        opam install -y ./ ./libs/ocaml-tree-sitter-core --deps-only
+    - name: Build semgrep-core
+      env:
+        ARTIFACT_PATH: /tmp/ocaml-build-artifacts.tgz
+      run: |
+        export PATH="${CYGWIN_ROOT_BIN}:${PATH}"
+        export TREESITTER_INCDIR=$(pwd)/libs/ocaml-tree-sitter-core/tree-sitter/include
+        export TREESITTER_LIBDIR=$(pwd)/libs/ocaml-tree-sitter-core/tree-sitter/lib
+
+        # cygwin cant handle rpath
+        for filename in $(find ./languages/ -name dune); do
+          grep -v rpath $filename > $filename.new
+          mv $filename.new $filename
+        done
+        for filename in $(find ./libs/ocaml-tree-sitter-core/ -name dune); do
+          grep -v rpath $filename > $filename.new
+          mv $filename.new $filename
+        done
+
+        # TODO: build everything once osemgrep_cli_interactive has been moved to pro
+        opam exec -- dune build src/main
+
+        tar czvf ${ARTIFACT_PATH} _build/default/src/main/Main.exe \
```

Comment:
maybe add a TODO saying we should find a way at some point to statically link semgrep for windows.
They managed to do it for the opam.exe binary for windows they distribute now.